sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -298,7 +304,10 @@ private void decodeDictionaryIds(
             // TODO: Convert dictionary of Binaries to dictionary of Longs
             if (!column.isNullAt(i)) {
               Binary v = dictionary.decodeToBinary(dictionaryIds.getDictId(i));
-              column.putLong(i, ParquetRowConverter.binaryToSQLTimestamp(v));
+              long rawTime = ParquetRowConverter.binaryToSQLTimestamp(v);
+              long adjTime =
+                  convertTz == null ? rawTime : DateTimeUtils.convertTz(rawTime, convertTz, UTC);
+              column.putLong(i, adjTime);
```

Comment:
we also need to update `readBinaryBatch`

Reply:
oh good point.   I suppose to get test coverage for this, I'd have to try to generate a parquet file without dictionary encoding from impala ...